osu.Game/Overlays/Profile/Sections/Recent/DrawableRecentActivity.cs

```diff
@@ -0,0 +1,165 @@
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+using osu.Game.Screens.Select.Leaderboards;
+
+namespace osu.Game.Overlays.Profile.Sections.Recent
+{
+    public class DrawableRecentActivity : DrawableProfileRow
+    {
+        private APIAccess api;
+
+        private readonly RecentActivity activity;
+
+        private LinkFlowContainer content;
+
+        public DrawableRecentActivity(RecentActivity activity)
+        {
+            this.activity = activity;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(APIAccess api)
+        {
+            this.api = api;
+
+            LeftFlowContainer.Padding = new MarginPadding { Left = 10, Right = 160 };
+
+            LeftFlowContainer.Add(content = new LinkFlowContainer
+            {
+                AutoSizeAxes = Axes.Y,
+                RelativeSizeAxes = Axes.X,
+            });
+
+            RightFlowContainer.Add(new OsuSpriteText
+            {
+                Text = activity.CreatedAt.LocalDateTime.ToShortDateString(),
+                Anchor = Anchor.TopRight,
+                Origin = Anchor.TopRight,
+                Font = "Exo2.0-RegularItalic",
+                TextSize = 12,
+                Colour = OsuColour.Gray(0xAA),
+            });
+
+            var formatted = createMessage();
+
+            content.AddLinks(formatted.Text, formatted.Links);
+        }
+
+        protected override Drawable CreateLeftVisual()
+        {
+            switch (activity.Type)
+            {
+                case RecentActivityType.Rank:
+                    return new DrawableRank(activity.ScoreRank)
+                    {
+                        RelativeSizeAxes = Axes.Y,
+                        Width = 60,
+                        FillMode = FillMode.Fit,
+                    };
+
+                case RecentActivityType.Achievement:
+                    return new MedalIcon(activity.Achievement.Slug)
+                    {
+                        RelativeSizeAxes = Axes.Y,
+                        Width = 60,
+                        FillMode = FillMode.Fit,
+                    };
+
+                default:
+                    return new Container
+                    {
+                        RelativeSizeAxes = Axes.Y,
+                        Width = 60,
+                        FillMode = FillMode.Fit,
+                    };
+            }
+        }
+
+        private string toAbsoluteUrl(string url) => $"{api.Endpoint}{url}";
+
+        private MessageFormatter.MessageFormatterResult createMessage()
+        {
+            string userLinkTemplate = $"[{toAbsoluteUrl(activity.User?.Url)} {activity.User?.Username}]";

```

Comment:
was hoping for local methods
ie.

`string userLinkTemplate() => $"...";`

This means it won't unnecessarily string interpolate the pieces which aren't required.